app/rest/metrics-collector/src/main/java/io/syndesis/rest/metrics/collector/MetricsCollector.java

```diff
@@ -0,0 +1,318 @@
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.PostConstruct;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.concurrent.CustomizableThreadFactory;
+import org.springframework.stereotype.Service;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonMappingException;
+
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.internal.readiness.Readiness;
+import io.syndesis.core.Json;
+import io.syndesis.dao.manager.DataManager;
+import io.syndesis.jsondb.GetOptions;
+import io.syndesis.jsondb.JsonDB;
+import io.syndesis.model.ListResult;
+import io.syndesis.model.integration.Integration;
+import io.syndesis.model.metrics.IntegrationMetricsSummary;
+
+@Service
+public class MetricsCollector implements Runnable, Closeable {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MetricsCollector.class);
+    private static final String HISTORY = "HISTORY";
+
+    private final KubernetesClient kubernetes;
+    private final RawMetricsHandler rawMetricsHandler;
+    private final DataManager dataManager;
+    private final JsonDB jsonDB;
+
+    private final ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor();
+    private final ExecutorService executor = Executors.newCachedThreadPool(new CustomizableThreadFactory("metrics-collector"));
+
+    @Autowired
+    public MetricsCollector(DataManager dataManager, JsonDB jsonDB, KubernetesClient kubernetes) {
+        this.dataManager = dataManager;
+        this.jsonDB = jsonDB;
+        this.kubernetes = kubernetes;
+        this.rawMetricsHandler = new PersistRawMetrics(jsonDB);
+    }
+
+
+    @PostConstruct
+    @SuppressWarnings("FutureReturnValueIgnored")
+    public void open() {
+        LOGGER.info("Starting metrics collector.");
+        scheduler.scheduleAtFixedRate(this, 10, 5, TimeUnit.SECONDS);
+    }
+
+
+    @Override
+    public void close() throws IOException {
+        LOGGER.info("Stopping metrics collector.");
+        close(scheduler);
+        close(executor);
+    }
+
+
+    @Override
+    public void run() {
+        LOGGER.info("Collecting metrics for active integration pods.");
+        try {
+            List<Pod> integrationPodList = kubernetes.pods().withLabel("integration").list().getItems();
```

Comment:
Not sure whether this is relevant to this PR but shouldn't we used namespaced labels ?

Reply:
If you are looking for live pods maybe a filter on the status would be useful here.

Reply:
I had trouble getting the `status.Phase` to work as a filter. There's a chance the client is not handling that properly but I am not 100% sure about that.